chain/bitcoind_conn.go

```diff
@@ -268,205 +234,32 @@ func (c *BitcoindConn) Stop() {
 	c.wg.Wait()
 }
 
-// blockEventHandler reads raw blocks events from the ZMQ block socket and
-// forwards them along to the current rescan clients.
-//
-// NOTE: This must be run as a goroutine.
-func (c *BitcoindConn) blockEventHandler() {
-	defer c.wg.Done()
-
-	log.Info("Started listening for bitcoind block notifications via ZMQ "+
-		"on", c.zmqBlockConn.RemoteAddr())
-
-	// Set up the buffers we expect our messages to consume. ZMQ
-	// messages from bitcoind include three parts: the command, the
-	// data, and the sequence number.
-	//
-	// We'll allocate a fixed data slice that we'll reuse when reading
-	// blocks from bitcoind through ZMQ. There's no need to recycle this
-	// slice (zero out) after using it, as further reads will overwrite the
-	// slice and we'll only be deserializing the bytes needed.
-	var (
-		command [len(rawBlockZMQCommand)]byte
-		seqNum  [seqNumLen]byte
-		data    = make([]byte, maxRawBlockSize)
-	)
+// sendBlockToClients is used to notify all rescan clients of a new block.
+func (c *BitcoindConn) sendBlockToClients(block *wire.MsgBlock) {
+	c.rescanClientsMtx.Lock()
+	defer c.rescanClientsMtx.Unlock()
 
-	for {
-		// Before attempting to read from the ZMQ socket, we'll make
-		// sure to check if we've been requested to shut down.
+	for _, client := range c.rescanClients {
 		select {
+		case client.blockNtfns <- block:
```

Comment:
Are these channels always buffered?

Reply:
no they are not. But they were not before either and this commit is meant to just be a refactor (see `zmqTxNtfns` and `zmqBlockNtfns`).
Should I change them to be buffered?

Reply:
> no they are not. But they were not before either and this commit is meant to just be a refactor (see zmqTxNtfns and zmqBlockNtfns).

Interesting, I don't think it matter as much in the polling implementation, but for the zmq implementation, buffered channels here might help to mitigate some of the issues w/ the zmq high water mark setting in bitcond.

Reply:
I think the area that would matter more in this case is the transaction notifications: at times the mempool arrival rate can be somewhat high.